src/main/java/com/conveyal/r5/analyst/cluster/AnalystWorker.java

```diff
@@ -212,25 +229,50 @@ public AnalystWorker(Properties config, TransportNetworkCache cache) {
         s3.setRegion(Region.getRegion(Regions.fromName(ec2info.region)));
     }
 
+    /**
+     * Consider shutting down if enough time has passed.
+     * The strategy depends on the fact that we are billed by the hour for EC2 machines.
+     * We only need to consider shutting them down once an hour. Leaving them alive for the rest of the hour provides
+     * immediate compute capacity in case someone starts another job.
+     * So a few minutes before each hour of uptime has elapsed, we check how long the machine has been idle.
+     */
+    public void considerShuttingDown() {
+        long now = System.currentTimeMillis();
+        if (now > nextShutdownCheckTime && autoShutdown) {
+            // Check again exactly one hour later (assumes billing in one-hour increments)
+            nextShutdownCheckTime += 60 * 60 * 1000;
+            if (now > lastSinglePointTime + (SINGLE_KEEPALIVE_MINUTES * 60 * 1000) &&
```

Comment:
conversion to ms happens here, so above comments should be revised.

Reply:
Noted. Thanks.